integration/feature/assembly/repo/build.sc

```diff
@@ -0,0 +1,34 @@
+import mill._, scalalib._
+
+trait Setup extends ScalaModule {
+  def scalaVersion = "2.13.11"
+  def sources = T.sources(T.workspace / "src")
+  def ivyDeps = super.ivyDeps() ++ Agg(
+    ivy"com.lihaoyi::scalatags:0.8.2",
+    ivy"com.lihaoyi::mainargs:0.4.0",
+    ivy"org.apache.avro:avro:1.11.1"
+  )
+}
+trait ExtraDeps extends ScalaModule {
+  def ivyDeps = super.ivyDeps() ++ Agg(
+    ivy"dev.zio::zio:2.0.15",
+    ivy"org.typelevel::cats-core:2.9.0",
+    ivy"org.apache.spark::spark-core:3.4.0",
+    ivy"dev.zio::zio-metrics-connectors:2.0.8",
+    ivy"dev.zio::zio-http:3.0.0-RC2"
+  )
+}
+
+object noExe extends Module {
+  object small extends Setup {
+    override def prependShellScript: T[String] = ""
+  }
+  object large extends Setup with ExtraDeps {
+    override def prependShellScript: T[String] = ""
+  }
+}
+
+object exe extends Module {
+  object small extends Setup
+  object large extends Setup with ExtraDeps
+}
```

Comment:
Does this need to be an integration test? I feel like having it be a `scalalib` unit test should be sufficient. Integration tests are slow and flaky so better to avoid them unless truly necessary (e.g. testing launcher logic, bootstrap logic, script compilation, etc)

Reply:
No. Since we now identified the issue, it can be a unit test that just make sure it properly works.

Reply:
Changed.